test/test_proxy_tensor.py

```diff
@@ -808,9 +808,13 @@ def f(x):
         shape_env = self._test_dynamic(f, [(3, 4)], test_inputs)
         self.assertTrue(shape_env.evaluate_guards_for_args(torch.randn(4, 5)))
         self.assertFalse(shape_env.evaluate_guards_for_args(torch.randn(25, 5)))
-        # TODO: There should eventually be guards for contiguity, but they're
-        # not currently being done yet
-        assert len(shape_env.guards) == 1, "\n" + shape_env.format_guards()
+        self.assertExpectedInline(shape_env.format_guards(), """\
+ - Eq(t0.size(0), s0)
+ - Eq(t0.size(1), s1)
+ - Eq(t0.storage_offset(), 0)
+ - Eq(t0.stride(0), s1)
+ - Eq(t0.stride(1), 1)
```

Comment:
Oh I guess this is a cost of always 0/1 specializing - even if our function happened to not use storage_offset/stride in a meaningful way, we still specialize and generate guards for it.

Curiosity question: does inductor generate a meaningfully different kernel for 0/1 strides/storage_offset (so specializing in cases where it seems unnecessary like in this function might still be profitable)?

Reply:
I think we should not zero specialize storage offset. Stride specialization is profitable though, it corresponds to assuming contiguity of inputs.